Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -22,18 +32,17 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
 
     override var profile: Profile! {
         didSet {
-            // Until we have something useful to show for desktop bookmarks,
-            // only show mobile bookmarks.
-            // Note that we also need to build a similar kind of virtual hierarchy
-            // to what we have on Android.
-            profile.bookmarks.modelForFolder(BookmarkRoots.MobileFolderGUID, success: self.onNewModel, failure: self.onModelFailure)
-            // profile.bookmarks.modelForRoot(self.onNewModel, failure: self.onModelFailure)
+            // Get all the bookmarks split by folders
+             profile.bookmarks.modelForRoot(self.onNewModel, failure: self.onModelFailure)
```

Comment:
@rnewman: I'm having a problem with traversal of the folder hierarchy in navigation after rebasing this on your bookmarking code.

While `profile.bookmarks.modelForFolder(BookmarkRoots.MobileFolderGUID).upon(onModelFetched)` provides me with the correct model from `MergedSQLiteBookmarks` (treating `mobile___` as root), the modelFactory that is returned for that model is a `SQLiteBookmarkMirrorStorage`, rather than a `MergedSQLiteBookmarks`. This mean that when I traverse back up the folder structure and fetch root again, it fetches the root from `SQLiteBookmarkMirrorStorage` which is empty. 

Any pointer on how this can be resolved, while retaining the correct `SQLiteBookmarkMirrorStorage` functionality?